co2eq/meal.js

```diff
@@ -46,8 +49,22 @@ export function carbonIntensity(ingredient) {
 Carbon emissions of an activity (in kgCO2eq)
 */
 export function carbonEmissions(activity) {
-  const { ingredients } = activity;
-  return ingredients
-    .map(k => carbonIntensity(k) * (MEAL_WEIGHT / 1000.0 / ingredients.length))
-    .reduce((a, b) => a + b, 0);
+  const {
+    ingredients, costEuros, purchaseCategory, passengerCount,
+  } = activity;
+  if (ingredients && Object.keys(ingredients).length > 0) {
+    return ingredients
+      .map(k => carbonIntensity(k) * (MEAL_WEIGHT / 1000.0 / ingredients.length))
+      .reduce((a, b) => a + b, 0);
+  }
+  if (costEuros) {
+    let intensity;
+    if (purchaseCategory) {
+      intensity = purchaseCarbonIntensity(purchaseCategory);
+    } else {
+      intensity = purchaseCarbonIntensity(PURCHASE_CATEGORY_FOOD_RESTAURANT);
+    }
+    return (intensity * costEuros) / (passengerCount || 1);
+  }
+  return null;
```

Comment:
Returning null will silently fail. We could potentially raise an error here so we get it in Sentry as this should in principle never happen.